### PR TITLE
doc/cephadm: core dumps

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Signed-off-by: Sebastian Wagner <sewagner@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests
- Teuthology
  - [ ] Completed teuthology run
  - [x] No teuthology test necessary (e.g., documentation)

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
https://ceph--43969.org.readthedocs.build/en/43969/cephadm/troubleshooting/#core-dumps 

(still building...)

File: doc/cephadm/troubleshooting.rst [hunk 1]
```diff
@@ -338,3 +338,33 @@ Deploy the daemon::
...
+
+.. prompt:: bash #
+
+  ulimit -c unlimited
+
+core dumps will now be written to ``/var/lib/systemd/coredump``.
+
+.. note::
+
+  core dumps are not namespaced by the kenel, which means
```
Comment (adk3798) [inline doc/cephadm/troubleshooting.rst:354]:
```suggestion
  core dumps are not namespaced by the kernel, which means
```

File: doc/cephadm/troubleshooting.rst [hunk 2]
```diff
@@ -338,3 +338,33 @@ Deploy the daemon::
...
+  they will be written to ``/var/lib/systemd/coredump`` on
+  the container host. 
+
+Now, wait for the creash to happen again. (To simulate the crash of a daemon, run e.g. ``killall -3 ceph-mon``)
```
Comment (adk3798) [inline doc/cephadm/troubleshooting.rst:358]:
```suggestion
Now, wait for the crash to happen again. (To simulate the crash of a daemon, run e.g. ``killall -3 ceph-mon``)
```

**Review by adk3798 — APPROVED** [review]

File: doc/cephadm/troubleshooting.rst [hunk 3]
```diff
@@ -338,3 +338,33 @@ Deploy the daemon::
...
+  core dumps are not namespaced by the kernel, which means
+  they will be written to ``/var/lib/systemd/coredump`` on
+  the container host. 
+
+Now, wait for the crash to happen again. (To simulate the crash of a daemon, run e.g. ``killall -3 ceph-mon``)
+
+Install debug packages by entering the cephadm shell and install ``ceph-debuginfo``::
+
+  # cephadm shell --mount /var/lib/systemd/coredump
+  [ceph: root@host1 /]# dnf install ceph-debuginfo gdb zstd
```
Comment (mgfritch) [inline doc/cephadm/troubleshooting.rst:363]:
```suggestion
  [ceph: root@host1 /]# dnf install ceph-mon-debuginfo gdb zstd
```

**Review by mgfritch — APPROVED** [review]

File: doc/cephadm/troubleshooting.rst (hunk 3 above)
Comment (sebastian-philipp) [reply doc/cephadm/troubleshooting.rst:363]: actually ceph-debuginfo works!

Comment (mgfritch) [reply doc/cephadm/troubleshooting.rst:363]: I didn't work for me using the upstream container image (quay.io/ceph/ceph:v16), which points at rpms from download.ceph.com ...

Comment (sebastian-philipp) [reply doc/cephadm/troubleshooting.rst:363]:
```
➜  cephadm git:(core) ✗ sudo ./cephadm shell --fsid ea6e85ae-47a6-11ec-98ba-482ae35a5fbb -c /etc/ceph/ceph.conf -k /etc/ceph/ceph.client.admin.keyring --mount /var/lib/systemd/coredump
Using recent ceph image quay.ceph.io/ceph-ci/ceph@sha256:9a0a72a98cd5ee02e66724987659ecb97332b5ec1eb32f3d284ac8a307bcb244
[ceph: root@sebastians-laptop /]# ls /mnt/coredump/
core.ceph-mon.167.87191dd15bab45c5bcfcd65ef6cce18e.3609467.1637155980000000.zst  core.python.1000.87191dd15bab45c5bcfcd65ef6cce18e.2868481.1637063059000000.zst
[ceph: root@sebastians-laptop /]# dnf install ceph-debuginfo gdb
Apache Arrow for CentOS 8 - x86_64                                                                                                                                                  82 kB/s | 156 kB     00:01    
CentOS Linux 8 - AppStream                                                                                                                                                         5.1 MB/s | 7.9 MB     00:01    
CentOS Linux 8 - BaseOS                                                                                                                                                            5.9 MB/s | 2.6 MB     00:00    
CentOS Linux 8 - Extras                                                                                                                                                            8.1 kB/s |  10 kB     00:01    
Copr repo for python-scikit-learn owned by tchaikov                                                                                                                                 46 kB/s |  20 kB     00:00    
Copr repo for python3-asyncssh owned by tchaikov                                                                                                                                   8.1 kB/s | 3.5 kB     00:00    
Ceph packages for x86_64                                                                                                                                                            47 kB/s |  75 kB     00:01    
Ceph noarch packages                                                                                                                                                               9.9 kB/s | 9.1 kB     00:00    
Ceph source packages                                                                                                                                                               1.5 kB/s | 1.4 kB     00:00    
ceph-iscsi packages for x86_64                                                                                                                                                     187  B/s | 257  B     00:01    
ceph-iscsi noarch packages                                                                                                                                                         2.4 kB/s | 2.2 kB     00:00    
ceph-iscsi source packages                                                                                                                                                         287  B/s | 257  B     00:00    
Extra Packages for Enterprise Linux 8 - x86_64                                                                                                                                     4.8 MB/s |  11 MB     00:02    
Extra Packages for Enterprise Linux Modular 8 - x86_64                                                                                                                             2.1 MB/s | 958 kB     00:00    
ganesha                                                                                                                                                                             96 kB/s |  28 kB     00:00    
tcmu-runner packages for x86_64                                                                                                                                                    2.1 kB/s | 2.9 kB     00:01    
tcmu-runner noarch packages                                                                                                                                                        284  B/s | 257  B     00:00    
tcmu-runner source packages                                                                                                                                                        280  B/s | 257  B     00:00    
Dependencies resolved.
===================================================================================================================================================================================================================
 Package                                            Architecture                              Version                                                           Repository                                    Size
===================================================================================================================================================================================================================
Installing:
 ceph-debuginfo                                     x86_64                                    2:17.0.0-9054.ge43a888e.el8                                       Ceph                                          25 M
 gdb                                                x86_64                                    8.2-16.el8                                                        appstream                                    298 k
Installing dependencies:
 cpp                                                x86_64                                    8.5.0-4.el8_5                                                     appstream                                     10 M
 gc                                                 x86_64                                    7.6.4-3.el8                                                       appstream                                    109 k
 gcc                                                x86_64                                    8.5.0-4.el8_5                                                     appstream                                     23 M
 gdb-headless                                       x86_64                                    8.2-16.el8                                                        appstream                                    3.7 M
 glibc-devel                                        x86_64                                    2.28-164.el8                                                      baseos                                       1.0 M
 glibc-headers                                      x86_64                                    2.28-164.el8                                                      baseos                                       480 k
 guile                                              x86_64                                    5:2.0.14-7.el8                                                    appstream                                    3.5 M
 isl                                                x86_64                                    0.16.1-6.el8                                                      appstream                                    841 k
 kernel-headers                                     x86_64                                    4.18.0-348.el8                                                    baseos                                       8.3 M
 libatomic_ops                                      x86_64                                    7.6.2-3.el8                                                       appstream                                     38 k
 libgomp                                            x86_64                                    8.5.0-4.el8_5                                                     baseos                                       206 k
 libipt                                             x86_64                                    1.6.1-8.el8                                                       appstream                                     50 k
 libmpc                                             x86_64                                    1.1.0-9.1.el8                                                     appstream                                     61 k
 libxcrypt-devel                                    x86_64                                    4.1.1-6.el8                                                       baseos                                        25 k
Installing weak dependencies:
 gcc-gdb-plugin                                     x86_64                                    8.5.0-4.el8_5                                                     appstream                                    118 k

Transaction Summary
===================================================================================================================================================================================================================
Install  17 Packages

Total download size: 77 M
Installed size: 257 M
Is this ok [y/N]: y
Downloading Packages:
(1/17): gc-7.6.4-3.el8.x86_64.rpm                                                                                                                                                  830 kB/s | 109 kB     00:00    
(2/17): gcc-gdb-plugin-8.5.0-4.el8_5.x86_64.rpm                                                                                                                                    1.3 MB/s | 118 kB     00:00    
(3/17): gdb-8.2-16.el8.x86_64.rpm                                                                                                                                                  1.2 MB/s | 298 kB     00:00    
(4/17): gdb-headless-8.2-16.el8.x86_64.rpm                                                                                                                                         2.6 MB/s | 3.7 MB     00:01    
(5/17): cpp-8.5.0-4.el8_5.x86_64.rpm                                                                                                                                               5.4 MB/s |  10 MB     00:01    
(6/17): isl-0.16.1-6.el8.x86_64.rpm                                                                                                                                                4.0 MB/s | 841 kB     00:00    
(7/17): libatomic_ops-7.6.2-3.el8.x86_64.rpm                                                                                                                                       1.0 MB/s |  38 kB     00:00    
(8/17): libipt-1.6.1-8.el8.x86_64.rpm                                                                                                                                              1.6 MB/s |  50 kB     00:00    
(9/17): libmpc-1.1.0-9.1.el8.x86_64.rpm                                                                                                                                            2.0 MB/s |  61 kB     00:00    
(10/17): glibc-devel-2.28-164.el8.x86_64.rpm                                                                                                                                       3.3 MB/s | 1.0 MB     00:00    
(11/17): glibc-headers-2.28-164.el8.x86_64.rpm                                                                                                                                     4.1 MB/s | 480 kB     00:00    
(12/17): guile-2.0.14-7.el8.x86_64.rpm                                                                                                                                             3.2 MB/s | 3.5 MB     00:01    
(13/17): libgomp-8.5.0-4.el8_5.x86_64.rpm                                                                                                                                          1.3 MB/s | 206 kB     00:00    
(14/17): libxcrypt-devel-4.1.1-6.el8.x86_64.rpm                                                                                                                                    1.0 MB/s |  25 kB     00:00    
(15/17): kernel-headers-4.18.0-348.el8.x86_64.rpm                                                                                                                                  5.9 MB/s | 8.3 MB     00:01    
(16/17): gcc-8.5.0-4.el8_5.x86_64.rpm                                                                                                                                              3.2 MB/s |  23 MB     00:07    
(17/17): ceph-debuginfo-17.0.0-9054.ge43a888e.el8.x86_64.rpm                                                                                                                       5.5 MB/s |  25 MB     00:04    
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                                                                              9.8 MB/s |  77 MB     00:07     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                                                           1/1 
  Installing       : libmpc-1.1.0-9.1.el8.x86_64                                                                                                                                                              1/17 
  Installing       : cpp-8.5.0-4.el8_5.x86_64                                                                                                                                                                 2/17 
  Running scriptlet: cpp-8.5.0-4.el8_5.x86_64                                                                                                                                                                 2/17 
  Installing       : libgomp-8.5.0-4.el8_5.x86_64                                                                                                                                                             3/17 
  Running scriptlet: libgomp-8.5.0-4.el8_5.x86_64                                                                                                                                                             3/17 
  Installing       : kernel-headers-4.18.0-348.el8.x86_64                                                                                                                                                     4/17 
  Running scriptlet: glibc-headers-2.28-164.el8.x86_64                                                                                                                                                        5/17 
  Installing       : glibc-headers-2.28-164.el8.x86_64                                                                                                                                                        5/17 
  Installing       : libxcrypt-devel-4.1.1-6.el8.x86_64                                                                                                                                                       6/17 
  Installing       : glibc-devel-2.28-164.el8.x86_64                                                                                                                                                          7/17 
  Running scriptlet: glibc-devel-2.28-164.el8.x86_64                                                                                                                                                          7/17 
  Installing       : libipt-1.6.1-8.el8.x86_64                                                                                                                                                                8/17 
  Installing       : libatomic_ops-7.6.2-3.el8.x86_64                                                                                                                                                         9/17 
  Installing       : gc-7.6.4-3.el8.x86_64                                                                                                                                                                   10/17 
  Installing       : guile-5:2.0.14-7.el8.x86_64                                                                                                                                                             11/17 
  Running scriptlet: guile-5:2.0.14-7.el8.x86_64                                                                                                                                                             11/17 
  Installing       : gdb-headless-8.2-16.el8.x86_64                                                                                                                                                          12/17 
  Installing       : isl-0.16.1-6.el8.x86_64                                                                                                                                                                 13/17 
  Running scriptlet: isl-0.16.1-6.el8.x86_64                                                                                                                                                                 13/17 
  Installing       : gcc-8.5.0-4.el8_5.x86_64                                                                                                                                                                14/17 
  Running scriptlet: gcc-8.5.0-4.el8_5.x86_64                                                                                                                                                                14/17 
  Installing       : gcc-gdb-plugin-8.5.0-4.el8_5.x86_64                                                                                                                                                     15/17 
  Running scriptlet: gcc-gdb-plugin-8.5.0-4.el8_5.x86_64                                                                                                                                                     15/17 
  Installing       : gdb-8.2-16.el8.x86_64                                                                                                                                                                   16/17 
  Installing       : ceph-debuginfo-2:17.0.0-9054.ge43a888e.el8.x86_64                                                                                                                                       17/17 
  Running scriptlet: guile-5:2.0.14-7.el8.x86_64                                                                                                                                                             17/17 
  Running scriptlet: ceph-debuginfo-2:17.0.0-9054.ge43a888e.el8.x86_64                                                                                                                                       17/17 
  Verifying        : cpp-8.5.0-4.el8_5.x86_64                                                                                                                                                                 1/17 
  Verifying        : gc-7.6.4-3.el8.x86_64                                                                                                                                                                    2/17 
  Verifying        : gcc-8.5.0-4.el8_5.x86_64                                                                                                                                                                 3/17 
  Verifying        : gcc-gdb-plugin-8.5.0-4.el8_5.x86_64                                                                                                                                                      4/17 
  Verifying        : gdb-8.2-16.el8.x86_64                                                                                                                                                                    5/17 
  Verifying        : gdb-headless-8.2-16.el8.x86_64                                                                                                                                                           6/17 
  Verifying        : guile-5:2.0.14-7.el8.x86_64                                                                                                                                                              7/17 
  Verifying        : isl-0.16.1-6.el8.x86_64                                                                                                                                                                  8/17 
  Verifying        : libatomic_ops-7.6.2-3.el8.x86_64                                                                                                                                                         9/17 
  Verifying        : libipt-1.6.1-8.el8.x86_64                                                                                                                                                               10/17 
  Verifying        : libmpc-1.1.0-9.1.el8.x86_64                                                                                                                                                             11/17 
  Verifying        : glibc-devel-2.28-164.el8.x86_64                                                                                                                                                         12/17 
  Verifying        : glibc-headers-2.28-164.el8.x86_64                                                                                                                                                       13/17 
  Verifying        : kernel-headers-4.18.0-348.el8.x86_64                                                                                                                                                    14/17 
  Verifying        : libgomp-8.5.0-4.el8_5.x86_64                                                                                                                                                            15/17 
  Verifying        : libxcrypt-devel-4.1.1-6.el8.x86_64                                                                                                                                                      16/17 
  Verifying        : ceph-debuginfo-2:17.0.0-9054.ge43a888e.el8.x86_64                                                                                                                                       17/17 

Installed:
  ceph-debuginfo-2:17.0.0-9054.ge43a888e.el8.x86_64     cpp-8.5.0-4.el8_5.x86_64                 gc-7.6.4-3.el8.x86_64                gcc-8.5.0-4.el8_5.x86_64              gcc-gdb-plugin-8.5.0-4.el8_5.x86_64    
  gdb-8.2-16.el8.x86_64                                 gdb-headless-8.2-16.el8.x86_64           glibc-devel-2.28-164.el8.x86_64      glibc-headers-2.28-164.el8.x86_64     guile-5:2.0.14-7.el8.x86_64            
  isl-0.16.1-6.el8.x86_64                               kernel-headers-4.18.0-348.el8.x86_64     libatomic_ops-7.6.2-3.el8.x86_64     libgomp-8.5.0-4.el8_5.x86_64          libipt-1.6.1-8.el8.x86_64              
  libmpc-1.1.0-9.1.el8.x86_64                           libxcrypt-devel-4.1.1-6.el8.x86_64      

Complete!

```

Comment (mgfritch) [reply doc/cephadm/troubleshooting.rst:363]: huh yeah there is a ceph-debuginfo package being build on master, but not pacific

Comment (sebastian-philipp) [reply doc/cephadm/troubleshooting.rst:363]: fascinating.

**Pull request merged**